gapis/gfxapi/resource.go

```diff
@@ -25,8 +25,11 @@ type ResourceMap map[Resource]id.ID
 
 // Resource represents an asset in a capture.
 type Resource interface {
-	// ResourceName returns the UI name for the resource.
-	ResourceName() string
+	// ResourceName returns the UI identity for the resource.
```

Comment:
> 1. Although the path is unique, there is no value in displaying it to the user.

I didn't mean to suggest that we'd present the path. I thought you were using `ResourceName` as an internal key, which you certainly shouldn't do as it's not unique.

> 2. We should display the GL texture/shader/program names. 

We always were

> 3. Separating the label from the name and letting the UI do the formatting is preferable, 1st normal form and all.

Okay, I can buy that.

> I changed the name to be numeric to allow sorting. 

I see that it's useful to sort by numerical value instead of string, but Vulkan is dealing with pointers. Decimal representations of pointers just seem wrong.

I think I'm mostly just offended with the concept of `ResourceName` returning an integer. "Name" may be the GL terminology, but it isn't for Vulkan.

How about:

```go
// ResourceHandle returns the API handle for the resource.
// For GL this is the GLuint object name, for Vulkan the pointer.
ResourceHandle() string

// ResourceLabel returns the annotated name for the resource.
ResourceLabel() string

// Order returns an integer used to sort the resources for presentation.
Order() int
```

We then use `Order` to sort the resources in [`gapis/resolve/resources.go`](https://github.com/google/gapid/blob/b1733c6bd337e1121973106b3cdb16c550e815b6/gapis/resolve/resources.go)